config2header: Move declaration of imapopts[] below declaration of
  struct imapopt_s. GCC doesn't like even external arrays without concrete
  types.
imap_err.et: Cyrus' bundled compile_et misses the line after a comment.

 fink/cyrus-makedirs                |  112 ++++++++++++++
 fink/cyrus-makedirs.8              |   46 ++++++
 fink/imapd.conf                    |  279 +++++++++++++++++++++++++++++++++++++
 imap/imap_err.et                   |    1 
 perl/imap/Makefile.PL              |    5 
 perl/sieve/managesieve/Makefile.PL |    3 
 tools/config2header                |    4 
 7 files changed, 449 insertions(+), 1 deletion(-)

Index: cyrus-imapd-2.2.12/tools/config2header
===================================================================
--- cyrus-imapd-2.2.12.orig/tools/config2header	2004-06-22 12:02:31.000000000 -0700
+++ cyrus-imapd-2.2.12/tools/config2header	2005-12-03 11:38:33.000000000 -0800
@@ -214,7 +214,6 @@ while (<STDIN>) {
 print HFILE <<EOF
   IMAPOPT_LAST
 };
-extern struct imapopt_s imapopts[];
 
 enum enum_value {
   IMAP_ENUM_ZERO = 0,
@@ -260,6 +259,9 @@ print HFILE "    const struct enum_optio
 
 print HFILE <<EOF
 };
+
+extern struct imapopt_s imapopts[];
+
 #endif /* INCLUDED_IMAPOPTIONS_H */
 EOF
     ;
Index: cyrus-imapd-2.2.12/imap/imap_err.et
===================================================================
--- cyrus-imapd-2.2.12.orig/imap/imap_err.et	2003-10-22 13:05:11.000000000 -0700
+++ cyrus-imapd-2.2.12/imap/imap_err.et	2005-12-03 11:51:40.000000000 -0800
@@ -119,6 +119,7 @@ ec IMAP_ANNOTATION_BADVALUE,
    "Invalid annotation value"
 
 # following used only proxy/db operations
+
 ec IMAP_BAD_SERVER,
    "Invalid server requested"
 
Index: cyrus-imapd-2.2.12/perl/imap/Makefile.PL
===================================================================
--- cyrus-imapd-2.2.12.orig/perl/imap/Makefile.PL	2003-10-22 11:50:18.000000000 -0700
+++ cyrus-imapd-2.2.12/perl/imap/Makefile.PL	2005-12-07 22:16:18.000000000 -0800
@@ -69,4 +69,9 @@ WriteMakefile( 
     'DEFINE'	=> '-DPERL_POLLUTE',     # e.g., '-DHAVE_SOMETHING' 
     'INC'	=> "-I../../lib $SASL_INC $OPENSSL_INC", 
     'EXE_FILES' => [cyradm],
+    'INSTALLSCRIPT' => "@PREFIX@/sbin",
+    'INSTALLARCHLIB' => "@PREFIX@/lib/perl5/5.8.6/darwin-thread-multi-2level",
+    'INSTALLSITEARCH' => "@PREFIX@/lib/perl5/5.8.6/darwin-thread-multi-2level",
+    'INSTALLSITEMAN1DIR' => "@PREFIX@/share/man/man1",
+    'INSTALLSITEMAN3DIR' => "@PREFIX@/share/man/man3",
 );
Index: cyrus-imapd-2.2.12/perl/sieve/managesieve/Makefile.PL
===================================================================
--- cyrus-imapd-2.2.12.orig/perl/sieve/managesieve/Makefile.PL	2005-12-07 21:24:15.000000000 -0800
+++ cyrus-imapd-2.2.12/perl/sieve/managesieve/Makefile.PL	2005-12-07 22:04:18.000000000 -0800
@@ -63,4 +63,7 @@ WriteMakefile(
     'LIBS'	=> ["$BDB_LIB $SASL_LIB $OPENSSL_LIB $LIB_RT -lssl -lcrypto"], 
     'DEFINE'	=> '-DPERL_POLLUTE',     # e.g., '-DHAVE_SOMETHING' 
     'INC'	=> "-I../lib/ -I../../../lib/ $SASL_INC $OPENSSL_INC",
+    'INSTALLARCHLIB' => "@PREFIX@/lib/perl5/5.8.6/darwin-thread-multi-2level",
+    'INSTALLSITEARCH' => "@PREFIX@/lib/perl5/5.8.6/darwin-thread-multi-2level",
+    'INSTALLSITEMAN3DIR' => "@PREFIX@/share/man/man3",
 );
Index: cyrus-imapd-2.2.12/fink/imapd.conf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ cyrus-imapd-2.2.12/fink/imapd.conf	2005-12-08 20:13:51.000000000 -0800
@@ -0,0 +1,279 @@
+# Debian/Fink Cyrus imapd.conf
+# See imapd.conf(5) for more information and more options
+
+# Configuration directory
+configdirectory: @PREFIX@/var/lib/cyrus
+
+# Which partition to use for default mailboxes
+defaultpartition: default
+partition-default: @PREFIX@/var/spool/cyrus/mail
+
+# News setup
+partition-news: @PREFIX@/var/spool/cyrus/news
+newsspool: @PREFIX@/var/spool/news
+
+# Alternate namespace
+# If enabled, activate the alternate namespace as documented in
+# @PREFIX@/share/doc/cyrus2-common/html/altnamespace.html, where an user's
+# subfolders are in the same level as the INBOX
+altnamespace: no
+
+# UNIX Hierarchy Convention
+# Set to yes, and cyrus will accept dots in names, and use the forward
+# slash "/" to delimit levels of the hierarchy. This is done by converting
+# internally all dots to "^", and all "/" to dots. So the "rabbit.holes"
+# mailbox of user "helmer.fudd" is stored in "user.elmer^fud.rabbit^holes"
+#
+# WARNING: This option does NOT apply to admin tools such as cyradm
+# (admins ONLY), reconstruct, quota, etc., NOR does it affect LMTP delivery
+# of messages directly to mailboxes via plus-addressing.
+# See also userprefix and sharedprefix on imapd.conf(5)
+unixhierarchysep: no
+
+# Munging illegal characters in headers
+# Headers of RFC2882 messages must not have characters with the 8th bit
+# set. However, too many badly-written MUAs generate this, including most
+# spamware.  Disable this if you want Cyrus to leave the crappage untouched
+# and you don't care that IMAP SEARCH won't work right anymore.
+#munge8bit: no
+
+# Forcing recipient user to lowercase
+# Cyrus 2.1 is case-sensitive.  If all your mail users are in lowercase, it is
+# probably a very good idea to set lmtp_downcase_rcpt to true.  The default is
+# to assume the user knows what he is doing, and not downcase anything.
+#lmtp_downcase_rcpt: yes
+
+# Uncomment the following and add the space-separated users who
+# have admin rights for all services.
+#admins: cyrus
+
+# Space-separated list of users that have lmtp "admin" status (i.e. that
+# can deliver email through TCP/IP lmtp) in addition to those in the
+# admins: entry above
+#lmtp_admins: postman
+
+# Space-separated list of users that have mupdate "admin" status, in
+# addition to those in the admins: entry above. Note that mupdate slaves and
+# backends in a Murder cluster need to autenticate against the mupdate master
+# as admin users.
+#mupdate_admins: mupdateman
+
+# Space-separated list of users that have imapd "admin" status, in
+# addition to those in the admins: entry above
+#imap_admins: cyrus
+
+# Space-separated list of users that have sieve "admin" status, in
+# addition to those in the admins: entry above
+#sieve_admins: cyrus
+
+# List of users and groups that are allowed to proxy for other users,
+# seperated by spaces.  Any user listed in this will be allowed to login
+# for any other user.  Like "admins:" above, you can have imap_proxyservers
+# and sieve_proxyservers.
+#proxyservers: cyrus
+
+# No anonymous logins
+allowanonymouslogin: no
+
+# Minimum time between POP mail fetches in minutes
+popminpoll: 1
+
+# If nonzero, normal users may create their own IMAP accounts by creating
+# the mailbox INBOX.  The user's quota is set to the value if it is positive,
+# otherwise the user has unlimited quota.
+autocreatequota: 0
+
+# umask used by Cyrus programs
+umask: 077
+
+# Sendmail binary location (used by sieve)
+sendmail: /usr/sbin/sendmail
+
+# If enabled, cyrdeliver will look for Sieve scripts in user's home
+# directories: ~user/.sieve.
+sieveusehomedir: false
+
+# If sieveusehomedir is false, this directory is searched for Sieve scripts.
+sievedir: @PREFIX@/var/spool/sieve
+
+# notifyd(8) method to use for "MAIL" notifications.  If not set, "MAIL"
+# notifications are disabled.  Valid methods are: null, log, zephyr
+#mailnotifier: zephyr
+
+# notifyd(8) method to use for "SIEVE" notifications.  If not set, "SIEVE"
+# notifications are disabled.  This method is only used when no method is
+# specified in the script.  Valid methods are null, log, zephyr, mailto
+#sievenotifier: zephyr
+
+# DRAC (pop-before-smtp, imap-before-smtp) support
+# Set dracinterval to the time in minutes to call DRAC while a user is
+# connected to the imap/pop services. Set to 0 to disable DRAC (default)
+# Set drachost to the host where the rpc drac service is running
+#dracinterval: 0
+#drachost: localhost
+
+# If enabled, the partitions will also be hashed, in addition to the hashing
+# done on configuration directories. This is recommended if one partition has a
+# very bushy mailbox tree.
+hashimapspool: true
+
+# Allow plaintext logins by default (SASL PLAIN)
+allowplaintext: yes
+
+# Force PLAIN/LOGIN authentication only
+# (you need to uncomment this if you are not using an auxprop-based SASL
+# mechanism.  saslauthd users, that means you!). And pay attention to
+# sasl_minimum_layer below, too.
+#sasl_mech_list: PLAIN
+
+# The minimum SSF that the server will allow a client to negotiate. A
+# value of 1 requires integrity protection; any higher value requires some
+# amount of encryption.
+#sasl_minimum_layer: 0
+
+# The maximum SSF that the server will allow a client to negotiate. A
+# value of 1 requires integrity protection; any higher value requires some
+# amount of encryption.
+#sasl_maximum_layer: 256
+
+# List of remote realms whose users may log in using cross-realm
+# authentications.  Seperate each realm name by a space. A cross-realm
+# identity is considered any identity returned by SASL with an "@" in it.
+#loginrealms:
+
+#
+# SASL library options (these are handled directly by the SASL libraries,
+# refer to SASL documentation for an up-to-date list of these)
+#
+
+# The mechanism(s) used by the server to verify plaintext passwords. Possible
+# values are "saslauthd", "auxprop", "pwcheck" and "alwaystrue".  They
+# are tried in order, you can specify more than one, separated by spaces.
+#
+# Do note that, since sasl will be run as user cyrus, you may have a lot of
+# trouble to set this up right.
+sasl_pwcheck_method: auxprop
+
+# What auxpropd plugins to load, if using sasl_pwcheck_method: auxprop
+# by default, all plugins are tried (which is probably NOT what you want).
+#sasl_auxprop_plugin: sasldb
+
+# If enabled, the SASL library will automatically create authentication secrets
+# when given a plaintext password. Refer to SASL documentation
+sasl_auto_transition: no
+
+#
+# SSL/TLS Options
+#
+
+# File containing the global certificate used for ALL services (imap, pop3,
+# lmtp, sieve)
+#tls_cert_file: @PREFIX@/etc/ssl/certs/cyrus-global.pem
+
+# File containing the private key belonging to the global server certificate.
+#tls_key_file: @PREFIX@/etc/ssl/private/cyrus-global.key
+
+# File containing the certificate used for imap. If not specified, the global
+# certificate is used.  A value of "disabled" will disable SSL/TLS for imap.
+#tls_imap_cert_file: @PREFIX@/etc/ssl/certs/cyrus-imap.pem
+
+# File containing the private key belonging to the imap-specific server
+# certificate.  If not specified, the global private key is used.  A value of
+# "disabled" will disable SSL/TLS for imap.
+#tls_imap_key_file: @PREFIX@/etc/ssl/private/cyrus-imap.key
+
+# File containing the certificate used for pop3. If not specified, the global
+# certificate is used.  A value of "disabled" will disable SSL/TLS for pop3.
+#tls_pop3_cert_file: @PREFIX@/etc/ssl/certs/cyrus-pop3.pem
+
+# File containing the private key belonging to the pop3-specific server
+# certificate.  If not specified, the global private key is used.  A value of
+# "disabled" will disable SSL/TLS for pop3.
+#tls_pop3_key_file: @PREFIX@/etc/ssl/private/cyrus-pop3.key
+
+# File containing the certificate used for lmtp. If not specified, the global
+# certificate is used.  A value of "disabled" will disable SSL/TLS for lmtp.
+#tls_lmtp_cert_file: @PREFIX@/etc/ssl/certs/cyrus-lmtp.pem
+
+# File containing the private key belonging to the lmtp-specific server
+# certificate.  If not specified, the global private key is used.  A value of
+# "disabled" will disable SSL/TLS for lmtp.
+#tls_lmtp_key_file: @PREFIX@/etc/ssl/private/cyrus-lmtp.key
+
+# File containing the certificate used for sieve. If not specified, the global
+# certificate is used.  A value of "disabled" will disable SSL/TLS for sieve.
+#tls_sieve_cert_file: @PREFIX@/etc/ssl/certs/cyrus-sieve.pem
+
+# File containing the private key belonging to the sieve-specific server
+# certificate.  If not specified, the global private key is used.  A value of
+# "disabled" will disable SSL/TLS for sieve.
+#tls_sieve_key_file: @PREFIX@/etc/ssl/private/cyrus-sieve.key
+
+# File containing one or more Certificate Authority (CA) certificates.
+#tls_ca_file: @PREFIX@/etc/ssl/certs/cyrus-imapd-ca.pem
+
+# Path to directory with certificates of CAs.
+tls_ca_path: @PREFIX@/etc/ssl/certs
+
+# The length of time (in minutes) that a TLS session will be cached for later
+# reuse.  The maximum value is 1440 (24 hours), the default.  A value of 0 will
+# disable session caching.
+tls_session_timeout: 1440
+
+# The list of SSL/TLS ciphers to allow.  The format of the string is described
+# in ciphers(1). THIS DISABLES THE WEAK 'FOR EXPORT' CRAP!
+tls_cipher_list: TLSv1:SSLv3:SSLv2:!NULL:!EXPORT:!DES:!LOW:@STRENGTH
+
+# Require a client certificate for ALL services (imap, pop3, lmtp, sieve).
+#tls_require_cert: false
+
+# Require a client certificate for imap ONLY.
+#tls_imap_require_cert: false
+
+# Require a client certificate for pop3 ONLY.
+#tls_pop3_require_cert: false
+
+# Require a client certificate for lmtp ONLY.
+#tls_lmtp_require_cert: false
+
+# Require a client certificate for sieve ONLY.
+#tls_sieve_require_cert: false
+
+#
+# Cyrus Murder cluster configuration
+#
+# Set the following options to the values needed for this server to
+# autenticate against the mupdate master server:
+# mupdate_server
+# mupdate_port
+# mupdate_username
+# mupdate_authname
+# mupdate_realm
+# mupdate_password
+# mupdate_retry_delay
+
+##
+## KEEP THESE IN SYNC WITH cyrus.conf
+##
+# Unix domain socket that lmtpd listens on.
+lmtpsocket: @PREFIX@/var/lib/cyrus/socket/lmtp
+
+# Unix domain socket that idled listens on.
+idlesocket: @PREFIX@/var/lib/cyrus/socket/idle
+
+# Unix domain socket that the new mail notification daemon listens on.
+notifysocket: @PREFIX@/var/lib/cyrus/socket/notify
+
+##
+## DEBUGGING
+##
+# Keep the hook disabled when it is not in use
+#
+# gdb Back-traces
+#debug_command: /usr/bin/gdb -batch -cd=/tmp -x @PREFIX@/lib/cyrus/get-backtrace.gdb @PREFIX@/lib/cyrus/bin/%s %d >/tmp/gdb-backtrace.cyrus.%1$s.%2$d <&- 2>&1 &
+#
+# system-call traces NOT AVAIL ON OS X
+#debug_command: /usr/bin/strace -tt -o /tmp/strace.cyrus.%s.%d -p %2$d <&- 2>&1 &
+#
+# library traces NOT AVAIL ON OS X
+#debug_command: /usr/bin/ltrace -tt -n 2 -o /tmp/ltrace.cyrus.%s.%d -p %2$d <&- 2>&1 &
Index: cyrus-imapd-2.2.12/fink/cyrus-makedirs
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ cyrus-imapd-2.2.12/fink/cyrus-makedirs	2005-12-08 19:42:27.000000000 -0800
@@ -0,0 +1,112 @@
+#!/bin/sh -e
+#
+#  cyrus-makedirs  -  Parses a Cyrus imap.d configuration file, and creates
+#                     the correct directory trees for all partitions
+#
+#  Copyright 2001,2002 by Henrique de Moraes Holschuh <hmh@debian.org.
+#  Released under the terms of the GNU General Public License (GPL) version 2
+#
+# $Id$
+
+# See lib/util.c, dir_hash_c for Cyrus' directory hashing
+# for the new hash style
+#HASHDIRS="A B C D E F G H I J K L M N O P Q R S T U V W"
+# for the old hash style
+HASHDIRS="a b c d e f g h i j k l m n o p q r s t u v w x y z"
+
+getconf () {
+	confvalue=`sed --silent -e "/^[[:blank:]]*$1:/ { \
+	s#^[[:blank:]]*$1:[[:blank:]]*## \
+	p
+	}" < "$CONF" | head -1`
+	result=${confvalue:-$2}
+}
+
+killsquat=0
+[ "$1" = "--cleansquat" ] && {
+	killsquat=1
+	shift
+}
+
+
+CONF="${1:-@PREFIX@/etc/imapd.conf}"
+[ -r "$CONF" ] || {
+	echo $0: unable to read configuration file $CONF. Aborting...
+	exit 1
+}
+
+getconf configdirectory @PREFIX@/var/lib/cyrus
+confdir="$result"
+mkdir -p "$confdir"
+[ -d "$confdir" ] || {
+	echo $0: $confdir is not a directory. Aborting...
+	exit 2
+}
+
+getconf sievedir @PREFIX@/etc/sieve
+sievedir="$result"
+
+getconf sieveusehomedir 0
+case "$result" in
+	1|t|true|yes|on) nosievedir=1
+			;;
+	*)		nosievedir=0
+			;;
+esac
+
+getconf hashimapspool 0
+case "$result" in
+	1|t|true|yes|on) hashspool=1
+			;;
+	*)		hashspool=0
+			;;
+esac
+
+# Partitions list
+partitions=`sed --silent -e "/^[[:blank:]]*partition-[[:alnum:]]\+:/ { \
+            s#^[[:blank:]]*partition-[[:alnum:]]\+:[[:blank:]]*## \
+	    p
+	    } " < "$CONF" | sort | uniq | xargs`
+
+# First, fix up the entire confdir subtree
+echo "Creating cyrus control directories in ${confdir}..."
+[ -d "$confdir" ] || mkdir -p "$confdir"
+chmod 750 "$confdir"
+for i in db proc socket log msg user quota; do
+    [ -d "$confdir/$i" ] || mkdir -p "$confdir/$i"
+    chmod 700 "$confdir/$i"
+done
+chmod 750 $confdir/socket
+for i in user quota ; do
+    for j in $HASHDIRS ; do
+    	[ -d "$confdir/$i/$j" ] || mkdir "$confdir/$i/$j"
+    done
+done
+chown -R cyrusimap:mail "$confdir"
+
+# Now, create the spool partitions
+for i in $partitions ; do
+	echo "Creating partition spool $i..."
+	[ -d "$i" ] || mkdir -p "$i"
+	chmod 750 "$i"
+	[ $hashspool -eq 1 ] && {
+	   for j in $HASHDIRS ; do
+	        [ -d "$i/$j" ] || mkdir "$i/$j"
+	   done
+	}
+	[ -d "$i/stage." ] || mkdir "$i/stage."
+	chown -R cyrusimap:mail "$i"
+	# and kill any squatter indexes
+	[ $killsquat -ne 0 ] && find "$i" -name 'cyrus.squat' -type f -exec rm -f "{}" \;
+done
+
+# And the sieve directory structure
+[ $nosievedir -eq 0 ] && {
+	[ -d "$sievedir" ] || mkdir "$sievedir"
+	chmod 755 "$sievedir"
+	for j in $HASHDIRS ; do
+		[ -d "$sievedir/$j" ] || mkdir "$sievedir/$j"
+		chmod 755 "$sievedir/$j"
+	done
+	chown -R cyrusimap:mail "$sievedir"
+}
Index: cyrus-imapd-2.2.12/fink/cyrus-makedirs.8
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ cyrus-imapd-2.2.12/fink/cyrus-makedirs.8	2005-12-08 19:32:35.000000000 -0800
@@ -0,0 +1,46 @@
+.\"                                      Hey, EMACS: -*- nroff -*-
+.\" First parameter, NAME, should be all caps
+.\" Second parameter, SECTION, should be 1-8, maybe w/ subsection
+.\" other parameters are allowed: see man(7), man(1)
+.TH CYRUS-MAKEDIRS 8 "2002-12-08"
+.\" Please adjust this date whenever revising the manpage.
+.\"
+.\" Some roff macros, for reference:
+.\" .nh        disable hyphenation
+.\" .hy        enable hyphenation
+.\" .ad l      left justify
+.\" .ad b      justify to both left and right margins
+.\" .nf        disable filling
+.\" .fi        enable filling
+.\" .br        insert line break
+.\" .sp <n>    insert n+1 empty lines
+.\" for manpage-specific macros, see man(7)
+.SH NAME
+cyrus-makedirs \- Sets up the Cyrus IMAPd system directories (spools)
+.SH SYNOPSIS
+.B cyrus-makedirs
+.RI [ configfile ]
+.br
+.SH DESCRIPTION
+This manual page documents briefly the
+.B cyrus-makedirs
+command.
+Cyrus-makedirs is the Debian version of Cyrus upstream's mkimap program.
+.PP
+You
+.B should
+call @PREFIX@/sbin/cyrus-makedirs when you change any imapd.conf(5) options
+that create or rename Cyrus mail spools/partitions.  It will be called
+automatically on package upgrades.
+.PP
+.\" TeX users may be more comfortable with the \fB<whatever>\fP and
+.\" \fI<whatever>\fP escape sequences to invode bold face and italics,
+.\" respectively.
+.SH OPTIONS
+You can optionally tell cyrus-makedirs the Cyrus imapd.conf config file
+to use.  @PREFIX@/etc/imapd.conf is assumed, if no parameters are given.
+.SH SEE ALSO
+.BR imapd.conf(5),
+.SH AUTHOR
+This manual page was written by Henrique de Moraes Holschuh <hmh@debian.org>,
+for the Debian GNU/Linux system (but may be used by others).
